docs/proposals/receiving_metadata.md

```diff
@@ -0,0 +1,49 @@
+# Support /api/v1/metadata in Cortex
+
+- Author: @gotjosh
+- Reviewers: @tomwilkie, @pracucci
+- Date: March 2020
+
+## Problem Statement
+Prometheus holds metric metadata alongside the contents of a scrape. This metadata (`HELP`, `TYPE`, `UNIT` and `METRIC_NAME`) enables [some Prometheus API](https://github.com/prometheus/prometheus/issues/6395) endpoints to output the metadata for integrations (e.g. [Grafana](https://github.com/grafana/grafana/pull/21124)) to consume it. 
+
+At the moment of writing, Cortex does not support the `api/v1/metadata` endpoint that Prometheus implements as metadata was never propagated via remote write. Recent [work is done in Prometheus](https://github.com/prometheus/prometheus/pull/6815/files) enables the propagation of metadata.
+
+With this in place, remote write integrations such as Cortex can now receive this data and implement the API endpoint. This result in Cortex users being able to enjoy a tiny bit more insight on their metrics.
+
+## Potential Solutions
+Before we delve into the solutions, let's set a baseline about how the data is received. This applies almost equally for the two.
+
+Metadata from Prometheus is sent in the same `WriteRequest` proto message that the samples use. It is part of a different field (#3 given #2 is already used), the data is a set identified by the metric name - that means it is aggregated across targets, and finally it is sent all at once. It is also important to note that this current process is an intermediary step. Eventually, metadata in a request will be sent alongside samples and only for those included. The solutions proposed, take this nuance into account to avoid coupling between the current and future state of Prometheus, and hopefully do something now that also works for the future. 
```

Comment:
> finally it is sent all at once

Can you clarify this? This assumes people are familiar with the Prometheus remote_write metadata protocol, I'd be a little more explicit because most of the reviewers of the proposal haven't yet seen it.

Reply:
Good point, done.